### PR TITLE
Refactor Language Support

### PR DESCRIPTION
**Pull request by KonradStanski**
This review centralizes the linguist language definitions.
I moves all language resolving methods to lib, separates the language syntax highlighting to a file which can be mapped to from the linguist metadata.
It also adds language syntax highlighting support for ~250 languages.

### CODE REVIEW

File: packages/web/package.json [hunk 1]
```diff
@@ -46,19 +49,42 @@
...
+    "codemirror-lang-prolog": "^0.1.0",
+    "codemirror-lang-r": "^0.1.1",
+    "codemirror-lang-sparql": "^2.0.0",
+    "codemirror-lang-spreadsheet": "^1.3.0",
+    "codemirror-lang-zig": "^0.1.0",
     "embla-carousel-auto-scroll": "^8.3.0",
     "embla-carousel-react": "^8.3.0",
     "escape-string-regexp": "^5.0.0",
     "fuse.js": "^7.0.0",
+    "graphql": "^16.9.0",
```
Comment (brendan-kellam) [inline packages/web/package.json:87]: do we need to take a dependency on graphql?

File: packages/web/src/lib/codemirrorLanguage.ts [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    "rust": rust(),
+    "sass": sass(),
+    "solidity": solidity,
+    "sparql": sparql(),
+    "spreadsheet": spreadsheet(),
+    "sql": sql(),
+    "svelte": svelte(),
+    "tsx": javascript({ jsx: true, typescript: true }),
+    "twig": twig(),
+    "typescript": javascript({ jsx: false, typescript: true }),
```
Comment (brendan-kellam) [inline packages/web/src/lib/codemirrorLanguage.ts:164]: nit: do we need a `javascript` key that defines `javascript({ jsx: false, typescript: false })`?

File: packages/web/package.json (hunk 1 above)
Comment (KonradStanski) [reply packages/web/package.json:87]: It's a peer dependency for qraphql syntax highlighting.

File: packages/web/src/lib/codemirrorLanguage.ts (hunk 2 above)
Comment (brendan-kellam) [reply packages/web/src/lib/codemirrorLanguage.ts:164]: ah I see what you mean - I see that `JavaScript` is being mapped to `jsx` so no worries, you can disregard my comment.

**Pull request merged**